src/cloudai/schema/test_template/nccl_test/kubernetes_json_gen_strategy.py

```diff
@@ -0,0 +1,262 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List
+
+from cloudai import JsonGenStrategy
+
+
+class NcclTestKubernetesJsonGenStrategy(JsonGenStrategy):
+    """JSON generation strategy for NCCL tests on Kubernetes systems."""
+
+    def gen_json(
+        self,
+        env_vars: Dict[str, str],
+        cmd_args: Dict[str, str],
+        extra_env_vars: Dict[str, str],
+        extra_cmd_args: str,
+        output_path: Path,
+        job_name: str,
+        num_nodes: int,
+        nodes: List[str],
+    ) -> Dict[Any, Any]:
+        final_env_vars = self._override_env_vars(self.default_env_vars, env_vars)
+        final_env_vars = self._override_env_vars(final_env_vars, extra_env_vars)
+        final_cmd_args = self._override_cmd_args(self.default_cmd_args, cmd_args)
+        final_num_nodes = self._determine_num_nodes(num_nodes, nodes)
+        job_spec = self._create_job_spec(
+            job_name, final_num_nodes, nodes, final_env_vars, final_cmd_args, extra_cmd_args
+        )
+
+        return job_spec
+
+    def _determine_num_nodes(self, num_nodes: int, nodes: List[str]) -> int:
+        """
+        Determine the final number of nodes based on provided nodes or num_nodes.
+
+        Args:
+            num_nodes (int): The initial number of nodes specified.
+            nodes (List[str]): A list of specific nodes provided.
+
+        Returns:
+            int: The final number of nodes to be used for the job.
+        """
+        return len(nodes) if nodes else num_nodes
+
+    def _create_job_spec(
+        self,
+        job_name: str,
+        final_num_nodes: int,
+        nodes: List[str],
+        env_vars: Dict[str, str],
+        cmd_args: Dict[str, str],
+        extra_cmd_args: str,
+    ) -> Dict[Any, Any]:
+        """
+        Create the MPIJob specification for running NCCL tests on a Kubernetes cluster.
+
+        Args:
+            job_name (str): The name of the Kubernetes job.
+            final_num_nodes (int): The final number of nodes determined.
+            nodes (List[str]): A list of specific nodes to run the job on.
+            env_vars (Dict[str, str]): A dictionary of environment variables for the job.
+            cmd_args (Dict[str, str]): A dictionary of command-line arguments for the NCCL test.
+            extra_cmd_args (str): Additional command-line arguments for the NCCL test.
+
+        Returns:
+            Dict[Any, Any]: A dictionary representing the Kubernetes MPIJob specification.
+        """
+        docker_image_url = "ghcr.io/coreweave/nccl-tests:12.4.1-cudnn-devel-ubuntu20.04-nccl2.21.5-1-85f9143"
```

Comment:
Why is this hardcoded here instead of reading from the cmd_args or test_template or its equivalent?

Reply:
Good catch! Thank you for identifying this issue. I will update it accordingly.